lib/src/time_expression.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright 2024 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![allow(missing_docs)]
+
+use chrono::{DateTime, TimeZone, Utc};
+use chrono_english::{parse_date_string, DateError, Dialect};
+
+/// Represents an expression to match dates and times.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum TimeExpression {
+    /// Represents all times at or after the given instant in time.
+    AtOrAfter(DateTime<Utc>),
```

Comment:
It's probably better to use `backend::Timestamp` and `MillisSinceEpoch` in public API.

Reply:
Replaced `DateTime<Utc>` with `backend::Timestamp`.